architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,206 @@
+
+The Applications RP provisions some resources synchronously, whereas for other resources whose creation can be time consuming, it has workers that enable async operation. To faciliate Async operation, the Applications RP adds the incoming request to an in-memory queue. Workers dequeue requests from the queue and process them. 
+
+Applications RP persists the radius resources metadata in etcd through API Server. For this, it interacts with Kubernetes Controller. 
+
+Below is a high level overview of various sub components in Applications RP
+![Applications RP](2024-10-applications-rp-threat-model/apprp.png) 
+
+### Implementation Details
+
+*Applications RP deploying a cloud Resource*
+
+For deploying an AWS / Azure resource, Applications RP sends a request to UCP. It stores neccsary provider configs in environment objects and provides them to UCP. The provider config for AWS is account id and region. The provider config for Azure is subscription ID and resource group. The credentials required for deployment are managed by UCP. 
+
+*Applications RP deploying a Kubernetes Resource* 
+
+For deploying a kubernetes resource, Applications RP sends a request to Controller. 
+
+*Applications RP deploying a Recipe*
+
+Recipe Engine sub component in Applications RP is responsible for managing and deploying recipes. It sends a request to controller to validate custom recipes. 
+It sends a request to UCP to deploy bicep recipes. It fetches the recipes to be deployed from the public module link for terraform recipe or the OCI registry for bicep recipe. It works with AWS/Azure to deploy terraform recipes.
+
+`/radius/deploy/Chart/templates/rp/configmaps.yaml` has the settings for all the providers used by Applications RP. 
+```
+storageProvider:
+      provider: "apiserver"
+      apiserver:
+        context: ""
+        namespace: "radius-system"
+    queueProvider:
+      provider: "apiserver"
+      name: "radius"
+      apiserver:
+        context: ""
+        namespace: "radius-system"
+    metricsProvider:
+      prometheus:
+        enabled: true
+        path: "/metrics"
+        port: 9090
+    profilerProvider:
+      enabled: true
+      port: 6060
+    secretProvider:
+      provider: kubernetes
+    server:
+      host: "0.0.0.0"
+      port: 5443
+    workerServer:
+      maxOperationConcurrency: 10
+      maxOperationRetryCount: 2
+    ucp:
+      kind: kubernetes
+    logging:
+      level: "info"
+      json: true
+    {{- if and .Values.global.zipkin .Values.global.zipkin.url }}
+    tracerProvider:
+      serviceName: "applications.core"
+      zipkin: 
+        url: {{ .Values.global.zipkin.url }}
+    {{- end }}
+```
+
+#### Use of Cryptography
+
+#### Storage of secrets
+
+Applications RP stores secrets for rendering some resources. For instance TLS termination in gateways needs storing TLS cert and key. For these cases, Applications RP uses kubernetes secrets. 
```

Comment:
Does this include all of the relevant cases?

Reply:
added several cases.